teuthology/suite/test/test_init.py

```diff
@@ -179,6 +180,29 @@ def fake_false(*args, **kwargs):
                     '--throttle', throttle,
                     '--machine-type', machine_type,
                 ])
+        
+    def test_machine_type_multi_error(self):
+        with pytest.raises(ScheduleFailError) as exc:
+            main([
+                    '--ceph', 'main',
+                    '--suite', 'suite_name',
+                    '--throttle', '3',
+                    '--machine-type', 'multi',
+                    '--dry-run'
+            ])
+        assert str(exc.value) == "Scheduling failed: 'multi' is not a valid machine_type. \
+Maybe you want 'gibba,smithi,mira' or similar"
+
+    def test_machine_type_none_error(self):
+        with pytest.raises(ScheduleFailError) as exc:
+            main([
+                    '--ceph', 'main',
+                    '--suite', 'suite_name',
+                    '--throttle', '3',
+                    '--machine-type', 'None',
+                    '--dry-run'
+            ])
+        assert str(exc.value) == "Scheduling failed: Must specify a machine_type"
```

Comment:
As well as unit testing these specific exceptions and making sure that the exception output is correct. We also want to make sure that with `--dry-run`, the function `util.schedule_fail()`
is not hitting `smtp.sendmail()` in `util.schedule_fail()`. We can do this by importing `teuthology.suite.util.schedule_fail`, patching it, and add something like `assert m_schedule_fail.smtp.sendmail.called == False`. You might have to mock more stuff but this should get you started.